gds-api-adapters.gemspec

```diff
@@ -23,6 +23,7 @@ Gem::Specification.new do |s|
   s.add_dependency 'lrucache', '~> 0.1.1'
   s.add_dependency 'rest-client', '~> 1.8.0'
   s.add_dependency 'rack-cache'
+  s.add_dependency 'uk_postcode', '2.1.0'
```

Comment:
Rubygems doesn't support optional dependencies so to do this we'd have to remove the dependency entirely.  We could make requiring `gds_api/mapit` fail if `uk_postcode` isn't available (it probably would anyway because the `require 'uk_postcode'` would raise a `LoadError` - but we might want to trap that and raise a nicer error.  We'd need to  put something in the readme explaining this though - it's not the best user experience.  If you're really making the mapit adapter optional, perhaps the solution is to extract the mapit adaptor into its own gem?